app/src/main/java/com/example/frolic/EventDetailsActivity.java

```diff
@@ -0,0 +1,209 @@
+        btnBack.setOnClickListener(v -> finish());
+
+        // Get eventId from Intent extras
+        eventId = getIntent().getStringExtra("eventId");
+
+        // Load event details from Firestore
+        if (eventId != null) {
+            loadEventDetails(eventId);
+        }
+    }
+
+    /**
+     * Loads event details from Firestore based on the provided eventId.
+     *
+     * @param eventId The ID of the event to load.
+     */
+    private void loadEventDetails(String eventId) {
+        DocumentReference eventRef = db.collection("events").document(eventId);
+        eventRef.get().addOnSuccessListener(documentSnapshot -> {
+            if (documentSnapshot.exists()) {
+                Event event = documentSnapshot.toObject(Event.class);
+                if (event != null) {
+                    tvEventName.setText(event.getEventName());
+                    tvEventDescription.setText(event.getEventDesc());
+                    tvEventDate.setText(event.getEventDate().toString());
+                    tvOrganizer.setText("Organized by: " + event.getOrganizerId());
+                }
+            }
+        });
+    }
+
+    /**
+     * Activity that displays a list of events an entrant is enrolled in.
+     */
+    public static class MyEventsActivity extends AppCompatActivity {
+
+        private static final String TAG = "MyEventsActivity";
+        private RecyclerView recyclerView;
+        private MyEventsAdapter adapter;
+        private ArrayList<Event> myEventsList = new ArrayList<>();
+        private Button btnBack;
+
+        // Firestore database instance
+        private FirebaseFirestore db = FirebaseFirestore.getInstance();
+        private CollectionReference eventsRef = db.collection("events");
+        private String entrantId;
+
+        @Override
+        protected void onCreate(Bundle savedInstanceState) {
+            super.onCreate(savedInstanceState);
+            setContentView(R.layout.activity_my_events);
+
+            // Get the entrant ID from the Intent
+            entrantId = getIntent().getStringExtra("entrantId");
+
+            // Initialize the back button and set click listener
+            btnBack = findViewById(R.id.btnBack);
+            btnBack.setOnClickListener(v -> {
+                finish(); // Go back to the previous activity
+            });
+
+            // Set up RecyclerView
+            recyclerView = findViewById(R.id.recyclerViewMyEvents);
+            recyclerView.setLayoutManager(new LinearLayoutManager(this));
+            adapter = new MyEventsAdapter(myEventsList);
+            recyclerView.setAdapter(adapter);
+
+            // Load events from Firestore
+            loadMyEventsFromFirestore();
+        }
+
+        /**
+         * Loads events the entrant is enrolled in from Firestore and updates the RecyclerView.
+         */
+        private void loadMyEventsFromFirestore() {
+            eventsRef.whereArrayContains("confirmedEntrants", entrantId)
```

Comment:
This line might be problematic, the confirmed entrants in Firebase is labeled as "entrantIds", which will return an array of String entrant ids